### PR TITLE
Add shallow clone and disable submodule options for scm utils

### PR DESCRIPTION
**Pull request by sleitner**

### CODE REVIEW

**Comment by imuchnik** [comment]
@sleitner either you are missing an example or I am not sure this will work

File: src/main/groovy/jenkins/automation/utils/ScmUtils.groovy [hunk 1]
```diff
@@ -24,8 +32,9 @@ class ScmUtils {
                         branch("*/tags/$version")
                     }
                     relativeTargetDir(repo.sub_directory)
+                    shallowClone(shallow)
```
Comment (imuchnik) [inline src/main/groovy/jenkins/automation/utils/ScmUtils.groovy]: shallowClone(repo.shallow) should be all you you need here

File: src/main/groovy/jenkins/automation/utils/ScmUtils.groovy [hunk 2]
```diff
@@ -9,6 +9,7 @@ class ScmUtils {
 
 
     static void project_repos(context, repos, use_versions = true) {
+        Boolean shallow
```
Comment (imuchnik) [inline src/main/groovy/jenkins/automation/utils/ScmUtils.groovy]: initialize it here.

File: src/main/groovy/jenkins/automation/utils/ScmUtils.groovy [hunk 3]
```diff
@@ -16,6 +17,13 @@ class ScmUtils {
                 def parsed_url = parsed_out_url[0]
 
                 def version = parsed_out_url[1]
+
+                if(repo.containsKey('shallow')) {
```
Comment (imuchnik) [inline src/main/groovy/jenkins/automation/utils/ScmUtils.groovy]: This puzzles me. An example of what you have in mind will be helpful.

File: src/main/groovy/jenkins/automation/utils/ScmUtils.groovy (hunk 1 above)
Comment (sleitner) [inline src/main/groovy/jenkins/automation/utils/ScmUtils.groovy]: repo.shallow will be null if the shallow key isn't provided, and shallowClone needs a boolean argument.

File: src/main/groovy/jenkins/automation/utils/ScmUtils.groovy (hunk 3 above)
Comment (sleitner) [inline src/main/groovy/jenkins/automation/utils/ScmUtils.groovy]: added one to example.md

File: src/main/groovy/jenkins/automation/utils/ScmUtils.groovy (hunk 1 above)
Comment (imuchnik) [inline src/main/groovy/jenkins/automation/utils/ScmUtils.groovy]: Initialize it to false (default) above, but I think when it is null it is defaulted to false by the plugin anyways.
http://jenkinsci.github.io/job-dsl-plugin/#method/javaposse.jobdsl.dsl.helpers.scm.GitContext.shallowClone

Comment (sleitner) [inline src/main/groovy/jenkins/automation/utils/ScmUtils.groovy]: Is this what you mean? 

```
        Boolean shallow = false
        shallow = repo.shallow
```

If so, then shallow is null and then `shallowClone(shallow)` receives a null argument and throws an error.

Comment (imuchnik) [inline src/main/groovy/jenkins/automation/utils/ScmUtils.groovy]: you can use Elvis operator and do something like
shallowClone(repo.shallow?:false)
and that should do it. 

?: ==> elvis ( one of my groovy favs)

Comment (sleitner) [inline src/main/groovy/jenkins/automation/utils/ScmUtils.groovy]: ah okay :+1:, one line then.

**Comment by sleitner** [comment]
should this be the default? why do we ever need the history of a repo to be deployed via jenkins?

**Comment by vickumar1981** [comment]
:+1:

**Comment by Scotchester** [comment]
I agree, shallow clone should be the default.

**Pull request merged**